internal/testing/runchecks.sh

```diff
@@ -43,6 +43,15 @@ else
   exit $result
 fi
 
+# Ensure that the code has no extra dependencies (including transitive
+# dependencies) that we're not already aware of by comparing with
+# ./internal/testing/alldeps
+#
+# Whenever project dependencies change, rerun ./internal/testing/listdeps.sh
+./internal/testing/listdeps.sh | diff ./internal/testing/alldeps - || {
```

Comment:
Unfortunately, this is only going to list the dependencies for the main module, not any of the submodules (see the `for` loop below). I do think this should run for all modules, and likely that means that `alldeps` should be a union of all the dependencies.